persistence/memorypersistence/aggregate.go

```diff
@@ -27,6 +27,24 @@ func (ds *dataStore) LoadAggregateMetaData(
 	}, nil
 }
 
+func (ds *dataStore) LoadAggregateSnapshot(
+	_ context.Context,
+	hk, id string,
+) (persistence.AggregateSnapshot, bool, error) {
+	ds.db.mutex.RLock()
+	defer ds.db.mutex.RUnlock()
+
+	key := instanceKey{hk, id}
+	if inst, ok := ds.db.aggregate.snapshot[key]; ok {
+		return inst, true, nil
+	}
+
+	return persistence.AggregateSnapshot{
+		HandlerKey: hk,
+		InstanceID: id,
+	}, false, nil
```

Comment:
```suggestion
	ss, ok := ds.db.aggregate.snapshot[key]; ok {
	return ss, ok, nil
```

This can be simplified somewhat because unlike the `AggregateMetaData` which is always "valid" even if it hadn't been persisted, the snapshot need only be valid if it's known to already exist. Suggestion also uses the `ss` var name as you'd used elsewhere.

Note also that the fail-case branch of the existing code is not yet covered by tests - mentioning that now in case it's not as obvious after applying my suggestion.